modules/dists/StencilDist.chpl

```diff
@@ -784,6 +857,133 @@ iter StencilImpl.activeTargetLocales(const space : domain = boundingBox) {
   }
 }
 
+// create a domain over a Stencil Distribution
+proc type Stencil.createDomain(
+  dom: domain,
+  targetLocales: [] locale = Locales,
+  fluff = makeZero(dom.rank, dom.idxType),
+  periodic = false
+) {
+  return dom dmapped Stencil(dom, targetLocales, fluff=fluff, periodic=periodic);
+}
+
+// create a domain over a Stencil Distribution constructed from a series of ranges
+proc type Stencil.createDomain(
+  rng: range...?k,
+  targetLocales: [] locale = Locales,
+  fluff: ?t = makeZero(k, int),
+  periodic = false
+) where isHomogeneousTupleType(t)
+{
+  return createDomain({(...rng)}, targetLocales, fluff, periodic);
+}
+
+proc type Stencil.createDomain(rng: range...?k) {
+  return createDomain({(...rng)}, fluff = makeZero(k, rng[0].idxType));
+}
+
+// create an array over a Stencil Distribution, default initialized
+proc type Stencil.createArray(
+  dom: domain,
+  type eltType,
+  targetLocales: [] locale = Locales,
+  fluff = makeZero(dom.rank, dom.idxType),
+  periodic = false
+) {
+  var D = createDomain(dom, targetLocales, fluff, periodic);
+  var A: [D] eltType;
+  return A;
+}
+
+// create an array over a Stencil Distribution, initialized with the given value or iterator
+proc type Stencil.createArray(
+  dom: domain,
+  type eltType,
+  initExpr: ?t,
+  targetLocales: [] locale = Locales,
+  fluff = makeZero(dom.rank, dom.idxType),
+  periodic = false
+) where isSubtype(t, _iteratorRecord) || isCoercible(t, eltType)
+{
+  var D = createDomain(dom, targetLocales, fluff, periodic);
+  var A: [D] eltType;
+  A = initExpr;
+  return A;
+}
+
+// create an array over a Stencil Distribution, initialized from the given array
+proc type Stencil.createArray(
+  dom: domain,
+  type eltType,
+  initExpr: [?arrayDom] ?arrayEltType,
+  targetLocales: [] locale = Locales,
+  fluff = makeZero(dom.rank, dom.idxType),
+  periodic = false
+) where dom.rank == arrayDom.rank && isCoercible(arrayEltType, eltType)
+{
+  for (d, ad, i) in zip(dom.dims(), arrayDom.dims(), 0..) do
+    if d.size != ad.size then halt("Domain size mismatch in 'Stencil.createArray' dimension " + i:string);
+  var D = createDomain(dom, targetLocales, fluff, periodic);
+  var A: [D] eltType;
+  A = initExpr;
+  return A;
+}
+
+// create an array over a Stencil Distribution constructed from a series of ranges, default initialized
+proc type Stencil.createArray(
+  rng: range...?k,
+  type eltType,
+  targetLocales: [] locale = Locales,
+  fluff: ?f = makeZero(k, int),
+  periodic = false
+) where isHomogeneousTupleType(f) {
+  return createArray({(...rng)}, eltType, targetLocales, fluff, periodic);
+}
+
+proc type Stencil.createArray(rng: range...?k, type eltType) {
+  return createArray({(...rng)}, eltType, fluff = makeZero(k, rng[0].idxType));
+}
+
+// create an array over a Stencil Distribution constructed from a series of ranges, initialized with the given value or iterator
+proc type Stencil.createArray(rng: range...?k, type eltType, initExpr: ?t)
```

Comment:
Here is another case, surely not the only one, where the `range` should be followed by `(?)` and I'd rather go wtihout `?k` if that works.